configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/FileDirectory.java

```diff
@@ -127,13 +127,13 @@ public FileReference addFile(File source) throws IOException {
             return addFile(source, fileReference, hash);
     }
 
-    public void delete(FileReference fileReference, Function<FileReference, Boolean> canBeDeleted) {
+    public void delete(FileReference fileReference, ApplicationRepository applicationRepository) {
         if (useLock.value())
             try (Lock lock = locks.lock(fileReference)) {
-                if (canBeDeleted.apply(fileReference))
-                    deleteDirRecursively(destinationDir(fileReference));
-                else
+                if (applicationRepository.isFileReferenceInUse(fileReference))
```

Comment:
It was reversed ?

Reply:
No, expression was evaluated immediately, not when calling `apply`.